src/components/Link/Link.scss

```diff
@@ -49,3 +49,10 @@
 .ms-Link {
   @include ms-Link;
 }
+
+// For button links we need to ensure by default that we inherit the font size. Otherwise
+// the button default font size will override.
+
+BUTTON.ms-Link {
+  font-size: inherit;
```

Comment:
I'm happy to remove the link font-size/weight stuff, but the important part is to reset BUTTON to default to inheriting the font... maybe it should be `font: inherit` instead of just size...